enigma-cli/src/main/java/cuchaz/enigma/command/InsertProposedMappingsCommand.java

```diff
@@ -29,42 +29,44 @@
 import javax.annotation.Nullable;
 
 public class InsertProposedMappingsCommand extends Command {
-	private static final String NAME = "insert-proposed-mappings";
-
 	public InsertProposedMappingsCommand() {
-		super(NAME);
+		super(Argument.INPUT_JAR.required(),
+				Argument.INPUT_MAPPINGS.required(),
+				Argument.MAPPING_OUTPUT.required(),
+				Argument.OUTPUT_MAPPING_FORMAT.required(),
+				Argument.ENIGMA_PROFILE.optional());
 	}
 
 	@Override
-	public String getUsage() {
-		return "<in jar> <source> <result> <result-format> [<profile>]";
+	public void run(String... args) throws Exception {
+		Path inJar = getReadablePath(this.getArg(args, 0));
+		Path source = getReadablePath(this.getArg(args, 1));
+		Path output = getWritablePath(this.getArg(args, 2));
+		String resultFormat = this.getArg(args, 3);
+		Path profilePath = getReadablePath(this.getArg(args, 4));
+
+		this.run(inJar, source, output, resultFormat, profilePath, null);
 	}
 
 	@Override
-	public boolean isValidArgument(int length) {
-		return length == 4 || length == 5;
+	public String getName() {
+		return "insert-proposed-mappings";
 	}
 
 	@Override
-	public void run(String... args) throws Exception {
-		Path inJar = getReadablePath(getArg(args, 0, "in jar", true));
-		Path source = getReadablePath(getArg(args, 1, "source", true));
-		Path output = getWritablePath(getArg(args, 2, "result", true));
-		String resultFormat = getArg(args, 3, "result-format", true);
-		Path profilePath = getReadablePath(getArg(args, 4, "profile", false));
-
-		run(inJar, source, output, resultFormat, profilePath, null);
+	public String getDescription() {
+		return "Adds all mappings proposed by the provided plugins into the given mappings.";
```

Comment:
I'd replace "provided plugins" with "plugins in the classpath, or the profile if present", or something along those lines